### PR TITLE
Feature/dump and diff

### PR DESCRIPTION
**Pull request by ghost**
Hi @grahame 
Changes here to to deal with diffing the prior and post states.
If testing with data shakes out any more issues I'll push up again.
Includes:
- changing to explicit set in tasks (didn't work on my local without this)
- addition of result field for ingest diff (didn't know if you wanted just to collate against 'xlsx' state or show separately, but can easily be changed)
- use of changes in bpa-ingest features branch where qc_linkage function now uses errors_callback.
- Haven't added/updated bpa-ingest sub-module in this PR as the changes are sitting in bpa-ingest branch atm.

### CODE REVIEW

File: Dockerfile-dev [hunk 1]
```diff
@@ -16,6 +16,13 @@ WORKDIR /app
...
+# Ensure changes to build doesn't re-install requirements
+WORKDIR /tmp/app
+COPY bpa-ingest /tmp/app/bpa-ingest
+COPY bpaworkflow/pyproject.toml bpaworkflow/poetry.lock /tmp/app/bpaworkflow/
+RUN cd /tmp/app/bpaworkflow && poetry install --no-root
+WORKDIR /app
+RUN rm -rf /tmp/app
 # Copy code and install the app
 COPY . /app
 RUN cd /app/bpaworkflow && poetry install
```
Comment (grahame) [inline Dockerfile-dev]: I don't understand what this change achieves – we still run the poetry install in /app, what does doing all this stuff in /tmp (which we then seem to delete) achieve?

**Review by grahame — APPROVED** [review]
Thanks Matt. Could you respond to my comment on the change to the Dockerfile before merging this?

File: Dockerfile-dev (hunk 1 above)
Comment (ghost) [reply Dockerfile-dev]: this was just poc code in local dev  (accidentally committed, sorry) for keeping versions from reinstalling -deleted now

**Comment by ghost** [comment]
ok @grahame 
Those extra error handlings I've added for:
- don't run if md5 or xlsx verifys fail
- catch all errors from prior or post package_and_resource_run as anything that should be ignored should be put into (ingest.py  's). This is a flag that maybe DEV intervention needed (msg is sucky but we can adjust later)
- gather anything from linkage_diff. If nothing returns it succeeds.

**Review by grahame — APPROVED** [review]

**Pull request merged**